dnf-behave-tests/features/environment.py

```diff
@@ -208,6 +208,41 @@ def get_cmd(self, context):
 
         return result
 
+    def get_microdnf_cmd(self, context):
+        result = ["microdnf"]
+        result.append(self.assumeyes_option)
+
+        # installroot can't be set via context for safety reasons
+        if self.installroot and self.installroot != "/":
+            result.append("--installroot={0}".format(self.installroot))
+            result.append("--noplugins")
+            result.append("--config={0}".format(self._get("config")))
+            result.append("--setopt=reposdir={0}".format(os.path.join(self.installroot, "etc/yum.repos.d")))
+            result.append("--setopt=varsdir={0}".format(os.path.join(self.installroot, "etc/dnf/vars")))
+            result.append("--setopt=cachedir={0}".format(os.path.join(self.installroot, "var/cache/yum")))
+
+        config = self._get("config")
+        if config:
+            result.append("--config={0}".format(config))
+
+        releasever = self._get("releasever")
+        if releasever:
+            result.append("--releasever={0}".format(releasever))
+
+        if not self.installroot or self.installroot == "/":
+            disable_plugins = self._get("disable_plugins")
+            if disable_plugins:
+                result.append("--disableplugin='*'")
+            plugins = self._get("plugins") or []
+            for plugin in plugins:
+                result.append("--enableplugin='{0}'".format(plugin))
+
+        setopts = self._get("setopts") or {}
+        for key, value in setopts.items():
+            result.append("--setopt={0}={1}".format(key, value))
```

Comment:
A lot of the code here is actually the same as in `get_cmd()` (since the switches are actually made compatible...), ideally you could share most of the code in a function called e.g. `generate_cmd_args()` and only add to that in the respective `get_cmd()` and `get_microdnf_cmd()`.